src/uptane/initialize.cc

```diff
@@ -128,28 +128,25 @@ bool Repository::initEcuKeys(const UptaneConfig& uptane_config) {
     primary_public = p11.getUriPrefix() + uptane_config.public_key_path;
     primary_private = p11.getUriPrefix() + uptane_config.private_key_path;
     std::string public_key_content;
-    if (!p11.readPublicKey(uptane_config.public_key_path, &public_key_content)) {
-      LOGGER_LOG(LVL_error, "Public key with id " << uptane_config.private_key_path << " was not found");
-      return false;
+    if (p11.readPublicKey(uptane_config.public_key_path, &public_key_content)) {
+      primary_public_id = Crypto::getKeyId(public_key_content);
+      setEcuKeysMembers(primary_public, primary_private, primary_public_id, kPkcs11);
+      return true;
     }
-    primary_public_id = Crypto::getKeyId(public_key_content);
-
-    setEcuKeysMembers(primary_public, primary_private, primary_public_id, kPkcs11);
-    return true;
+    if (!p11.generateRSAKeyPair(uptane_config.private_key_path)) return false;
```

Comment:
We need to make sure that `private_key_path` and `public_key_path` are the same in the pkcs#11 case. Or just use one of these variables everywhere.

Reply:
It is keypair, and when we ganarate it we need to pass the single id

Reply:
Yes You are right, I will add some checks